From 438166292fd86948adb8407ecb9fd9765b625598 Mon Sep 17 00:00:00 2001
From: Robert Shih <robertshih@google.com>
Date: Fri, 9 Feb 2018 12:19:35 -0800
Subject: [PATCH 1/5] M3UParser: detect variant streams without
 EXT-X-STREAM-INF

Bug: 68399439
Change-Id: I95207b40f23a5f927da7154f9a952046118b5cad
(cherry picked from commit f3e0afb82f104d6e9986779ba2cf548c6aab1092)
(cherry picked from commit d32af5db10f018219e0379f333c7f0452a4f7a31)
---
 media/libstagefright/httplive/M3UParser.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/httplive/M3UParser.cpp b/media/libstagefright/httplive/M3UParser.cpp
index 78a2dd0..1242c95 100644
--- a/media/libstagefright/httplive/M3UParser.cpp
+++ b/media/libstagefright/httplive/M3UParser.cpp
@@ -658,10 +658,13 @@ status_t M3UParser::parse(const void *_data, size_t size) {
         }
 
         if (!line.startsWith("#")) {
+            if (itemMeta == NULL) {
+                ALOGV("itemMeta == NULL");
+                return ERROR_MALFORMED;
+            }
             if (!mIsVariantPlaylist) {
                 int64_t durationUs;
-                if (itemMeta == NULL
-                        || !itemMeta->findInt64("durationUs", &durationUs)) {
+                if (!itemMeta->findInt64("durationUs", &durationUs)) {
                     return ERROR_MALFORMED;
                 }
                 itemMeta->setInt32("discontinuity-sequence",
-- 
2.7.4

